### ISSUE TITLE
UI

### ISSUE THREAD

**Issue by soulgin** [soulgin/Calendar-task-planner-app#30]
HTML for calendar integration.

**Comment by soulgin** [comment]
login: <!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Login</title>
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css" />
  </head>
  <body>
    <div class="container">
      <div class="row">
        <div class="col-md-6 offset-md-3">
          <h2 class="mt-4 mb-4">Login</h2>
          <form method="post" class="mb-4">
            {% csrf_token %}
            {{ form.as_p }}
            <button type="submit" class="btn btn-primary">Login</button>
          </form>
          <p>
            Don't have an account? <a href="{% url 'register' %}" class="btn btn-link">Register here</a>.
          </p>
        </div>
      </div>
    </div>
  </body>
</html>

**Comment by soulgin** [comment]
register:
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Register</title>
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css" />
  </head>
  <body>
    <div class="container">
      <div class="row">
        <div class="col-md-6 offset-md-3">
          <h2 class="mt-4 mb-4">Register</h2>
          <form method="post" class="mb-4">
            {% csrf_token %}
            {{ form.as_p }}
            <button type="submit" class="btn btn-primary">Register</button>
          </form>
          <p>Already have an account? <a href="{% url 'login' %}" class="btn btn-link">Login here</a>.</p>
        </div>
      </div>
    </div>
  </body>
</html>

**Comment by soulgin** [comment]
user task list: 
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>{{ category.name }} tasks</title>
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css" />
  </head>
  <body>
    <div class="container mt-5">
      <h2 class="text-center">{{ category.name }} tasks</h2>
      <br />
      <div class="mb-3">
        <a href="{% url 'logout' %}" class="btn btn-danger ml-2">logout</a> 
      </div>
      <table id="taskTable" class="table">
        <thead>
          <tr>
            <th>Task name</th>
            <th>Start Date</th>
            <th>Time Left</th>
            <th>
              Priority <button class="btn btn-link" onclick="sortByPriority()">&#x25B2;</button>
            </th> 
            <th>Actions</th> 
          </tr>
        </thead>
        <tbody>
          {% for task in tasks %}
            <tr>
              <td>{{ task.name }}</td>
              <td>{{ task.start_date }}</td>
              <td>
                <div id="countdown_{{ task.id }}" class="countdown-timer"></div>
              </td>
              <td>{{ task.priority }}</td>
              <td>
                <button type="button" class="btn btn-info btn-sm" data-toggle="modal" data-target="#taskModal_{{ task.id }}">Details</button> 
              </td>
            </tr>
          {% endfor %}
        </tbody>
      </table>
    </div>

    
    {% for task in tasks %}
      <div class="modal fade" id="taskModal_{{ task.id }}" tabindex="-1" role="dialog" aria-labelledby="taskModalLabel_{{ task.id }}" aria-hidden="true">
        <div class="modal-dialog" role="document">
          <div class="modal-content">
            <div class="modal-header">
              <h5 class="modal-title" id="taskModalLabel_{{ task.id }}">{{ task.name }}</h5>
              <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
            </div>
            <div class="modal-body">
              <p>
                <strong>Name:</strong> {{ task.name }}
              </p>
              <p>
                <strong>Category:</strong> {{ task.category.name }}
              </p>
              <p>
                <strong>Assigned to:</strong> {{ task.assigned_to.username }}
              </p>
              <p>
                <strong>Start Date:</strong> {{ task.start_date }}
              </p>
              <p>
                <strong>End Date:</strong> {{ task.end_date }}
              </p>
              <p>
                <strong>Priority:</strong> {{ task.priority }}
              </p>
              <p>
                <strong>Description:</strong> {{ task.description }}
              </p>
              <p>
                <strong>Location:</strong> {{ task.location }}
              </p>
              <p>
                <strong>Organizer:</strong> {{ task.organizer }}
              </p>
            </div>
            <div class="modal-footer">
              <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
            </div>
          </div>
        </div>
      </div>
    {% endfor %}

    
    <script>
    function sortByPriority() {
      var table, rows, switching, i, x, y, shouldSwitch;
      table = document.getElementById("taskTable");
      switching = true;
      while (switching) {
        switching = false;
        rows = table.getElementsByTagName("tr");
        for (i = 1; i < (rows.length - 1); i++) {
          shouldSwitch = false;
          x = rows[i].getElementsByTagName("td")[3]; // Index of Priority column
          y = rows[i + 1].getElementsByTagName("td")[3];
          if (parseInt(x.innerHTML) < parseInt(y.innerHTML)) {
            shouldSwitch = true;
            break;
          }
        }
        if (shouldSwitch) {
          rows[i].parentNode.insertBefore(rows[i + 1], rows[i]);
          switching = true;
        }
      }
    }

    function updateCountdownTimers() {
      {% for task in tasks %}
      var startDate = new Date('{{ task.start_date|date:"Y/m/d H:i:s" }}');
      var now = new Date();

      var timeDiff = startDate - now;

      if (timeDiff > 0) {
        var days = Math.floor(timeDiff / (1000 * 60 * 60 * 24));
        var hours = Math.floor((timeDiff % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
        var minutes = Math.floor((timeDiff % (1000 * 60 * 60)) / (1000 * 60));
        var seconds = Math.floor((timeDiff % (1000 * 60)) / 1000);

        document.getElementById('countdown_{{ task.id }}').innerText = days + "d " + hours + "h "
          + minutes + "m " + seconds + "s ";
      } else {
        document.getElementById('countdown_{{ task.id }}').innerText = "task has started";
      }
      {% endfor %}
    }

    // Call the updateCountdownTimers function every second to update the countdown timers in real-time
    setInterval(updateCountdownTimers, 1000);
  </script>

    
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.16.0/umd/popper.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.5.2/js/bootstrap.min.js"></script>
  </body>
</html>

**Comment by soulgin** [comment]
HOME:

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Task Manager</title>
</head>
<body>
    <h1>Welcome to the Task Manager</h1>
    <h2>Months</h2>
    <ul>
        {% for month in months %}
            <li>{{ month.month }}/{{ month.year }}</li>
        {% endfor %}
    </ul>

    <h2>Your Tasks</h2>
    <ul>
        {% for task in tasks %}
            <li>{{ task.description }} - {{ task.day }}/{{ task.month.month }}/{{ task.month.year }}</li>
        {% endfor %}
    </ul>
</body>
</html>
